pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/ReadBridge.java

```diff
@@ -14,58 +14,81 @@
 
 import java.io.*;
 import java.nio.charset.CharacterCodingException;
+import java.util.LinkedList;
 import java.util.zip.ZipException;
 
-/*
- * ReadBridge class creates appropriate accessor and resolver.
- * It will then create the correct output conversion
- * class (e.g. Text or GPDBWritable) and get records from accessor,
- * let resolver deserialize them and reserialize them using the
- * output conversion class.
- *
- * The class handles BadRecordException and other exception type
- * and marks the record as invalid for GPDB.
+/**
+ * ReadBridge class creates appropriate accessor and resolver. It will then
+ * create the correct output conversion class (e.g. Text or GPDBWritable) and
+ * get records from accessor, let resolver deserialize them and reserialize them
+ * using the output conversion class. <br>
+ * The class handles BadRecordException and other exception type and marks the
+ * record as invalid for HAWQ.
  */
 public class ReadBridge implements Bridge {
     ReadAccessor fileAccessor = null;
     ReadResolver fieldsResolver = null;
     BridgeOutputBuilder outputBuilder = null;
+    LinkedList<Writable> outputQueue = null;
 
-    private Log Log;
+    static private Log Log = LogFactory.getLog(ReadBridge.class);
 
-    /*
-     * C'tor - set the implementation of the bridge
+    /**
+     * C'tor - set the implementation of the bridge.
+     *
+     * @param protData input containing accessor and resolver names
+     * @throws Exception if accessor or resolver can't be instantiated
      */
     public ReadBridge(ProtocolData protData) throws Exception {
         outputBuilder = new BridgeOutputBuilder(protData);
-        Log = LogFactory.getLog(ReadBridge.class);
+        outputQueue = new LinkedList<Writable>();
         fileAccessor = getFileAccessor(protData);
         fieldsResolver = getFieldsResolver(protData);
     }
 
-    /*
-     * Accesses the underlying HDFS file
+    /**
+     * Accesses the underlying HDFS file.
      */
     @Override
     public boolean beginIteration() throws Exception {
         return fileAccessor.openForRead();
     }
 
-    /*
-     * Fetch next object from file and turn it into a record that the GPDB backend can process
+    /**
+     * Fetches next object from file and turn it into a record that the HAWQ
+     * backend can process.
      */
     @Override
     public Writable getNext() throws Exception {
-        Writable output;
+        Writable output = null;
```

Comment:
http://stackoverflow.com/a/16699632
Apparently this rule excludes local variables. They cannot be returned unless they were initialized.